sale_ux/views/sale_order_views.xml

```diff
@@ -21,6 +21,12 @@
             <button name="action_cancel" position="attributes">
                 <attribute name="states">draft,sent,sale,done</attribute>
             </button>
+            <field name="partner_id" position="after">
+                <field name="commercial_partner_id" invisible="1"/>
+            </field>
+            <field name="analytic_account_id" position="attributes">
+                <attribute name="domain">['|',('partner_id','=',False),('partner_id.commercial_partner_id','=',commercial_partner_id)]</attribute>
```

Comment:
Considere when the analytic account has not partner and when it have the commercial_partner_id but currently not taking into account the partner itself. 

Maybe you could use the next domain

 > ['|','|',('partner_id','=',False), ('partner_id','=', partner_id), 
 ('partner_id.commercial_partner_id','=',commercial_partner_id)]

Reply:
If partner is commercial_partner_id then no problem. partner_id.commercial_partner_id will be the same as partner_id. 
So, default domain is ok "sale_usability_delivery a sale_delivery_ux"